### PR TITLE
test(robot): add test strict local volume disabled revision conuter by default

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9318

#### What this PR does / why we need it:

add test strict local volume disabled revision conuter by default

### CODE REVIEW

**Review by khushboo-rancher — COMMENTED** [review]
LGTM, few minor comments.

Also, do we want to add the test skeleton also for https://github.com/longhorn/longhorn/issues/9279#issuecomment-2307833776?

File: e2e/libs/replica/crd.py [hunk 1]
```diff
@@ -33,10 +33,7 @@ def delete(self, volume_name, node_name):
             logging(
                 f"Deleting volume {volume_name} on node {node_name} replicas")
 
-        resp = self.get(volume_name, node_name)
-        assert resp != "", f"failed to get replicas"
```
Comment (roger-ryao) [inline e2e/libs/replica/crd.py:37]: Regarding the removal of `assert resp != "", f"failed to get replicas"`, could you explain the main reasoning behind removing this step? I don't quite understand it.

File: e2e/libs/volume/crd.py [hunk 2]
```diff
@@ -495,3 +497,7 @@ def upgrade_engine_image(self, volume_name, engine_image_name):
 
     def wait_for_engine_image_upgrade_completed(self, volume_name, engine_image_name):
         return Rest(self.node_exec).wait_for_engine_image_upgrade_completed(volume_name, engine_image_name)
+    def validate_volume_setting(self, volume_name, setting_name, value):
```
Comment (roger-ryao) [inline e2e/libs/volume/crd.py:500]: Add an extra line of separation in the function.

File: e2e/libs/replica/crd.py (hunk 1 above)
Comment (roger-ryao) [inline e2e/libs/replica/crd.py:37]: Regarding the removal of `assert resp != "", f"failed to get replicas"`, could you explain the main reasoning behind removing this step? I don't quite understand it.

**Review by khushboo-rancher — APPROVED** [review]
LGTM from my side.

@roger-ryao has few questions.

**Comment by khushboo-rancher** [comment]
@roger-ryao Could you take a look whenever get a chance?

**Review by roger-ryao — APPROVED** [review]
LGTM, but I noticed that @khushboo-rancher and I have similar questions.  If possible, could you also explain the reason for removing these two sections, so we can understand more?

https://github.com/longhorn/longhorn-tests/pull/2080#discussion_r1768129952
https://github.com/longhorn/longhorn-tests/pull/2080#discussion_r1761984542

**Pull request merged**